### PR TITLE
Disable /search and hide help item if not enabled

### LINKED ISSUE #4273: Disable `/search query` if search is not possible.
Search only works if sqlite `messageStorage` is enabled. Right now, when this is not enabled, you don't get the magnifying glass icon but `/search query` is still available. Searching for something opens up the search window but it gets stuck in the "searching.." state

Closing that window takes you back to the channel and reveals an error VUE ERROR: CANNOT READ PROPERTY 'SLICE' OF UNDEFINED:

I guess the sane solution would be to disable the `/search` command if it's not possible (either because logging to sqlite is disabled globally or for the user)

### PR DESCRIPTION
**Pull request by MaxLeiter**
Fixes #4273 and supersedes #4278.

Removes `/search` from the client side commands if it's not enabled and hides its help item

### CODE REVIEW

File: client/js/commands/search.js [hunk 1]
```diff
@@ -4,10 +4,15 @@ import store from "../store";
 import {router} from "../router";
 
 function input(args) {
+	if (!store.state.settings.searchEnabled) {
+		return true;
+	}
+
+	const {channel} = store.state.activeChannel;
```
Comment (itsjohncs) [inline client/js/commands/search.js:11]: This seems like a fairly random stylistic change to include here and doesn't seem necessary.

File: client/js/commands/search.js [hunk 2]
```diff
@@ -4,10 +4,15 @@ import store from "../store";
 import {router} from "../router";
 
 function input(args) {
+	if (!store.state.settings.searchEnabled) {
+		return true;
```
Comment (itsjohncs) [inline client/js/commands/search.js:8]: I wonder if just returning false here would be reasonable. We'd end up behaving as if the command is completely unknown, which seems reasonable given that we're stripping its existence from the UI while its disabled.

File: src/plugins/inputs/index.js [hunk 3]
```diff
@@ -47,7 +45,6 @@ const getCommands = () =>
 	Array.from(userInputs.keys())
 		.concat(Array.from(pluginCommands.keys()))
 		.map((command) => `/${command}`)
-		.concat(clientSideCommands)
```
Comment (itsjohncs) [inline src/plugins/inputs/index.js:50]: I like that we're not sending the client the commands it already knows now :).

**Review by itsjohncs — COMMENTED** [review]
Haven't tested this yet, but did a somewhat thorough look-through the code. Hopefully this is the sort of review you're after. I'm happy to focus solely on functionality instead of softer/opinionated qualities.

File: src/plugins/inputs/index.js (hunk 3 above)
Comment (MaxLeiter) [reply src/plugins/inputs/index.js:50]: This is now done in `client/js/socket-events/commands.js`, but thats a change from the previous PR and not strictly necessary.

**Comment by MaxLeiter** [comment]
@itsjohncs can't re-request review for some reason, but I addressed your comments.

File: client/js/autocompletion.js [hunk 4]
```diff
@@ -311,10 +311,19 @@ function completeNicks(word, isFuzzy) {
 	return users.filter((w) => !w.toLowerCase().indexOf(word));
 }
 
-function completeCommands(word) {
-	const words = constants.commands.slice();
+function getCommands() {
+	let cmds = constants.commands.slice();
 
-	return fuzzyGrep(word, words);
+	if (store.state.settings.searchEnabled === false) {
```
Comment (itsjohncs) [inline client/js/autocompletion.js:317]: why is this being compared to false rather than checking to see if its falsy?

**Comment by itsjohncs** [comment]
digging it, going to quickly dogfood it this evening and see if i can shake out any actual bugs.

**Review by itsjohncs — APPROVED** [review]
Shook the tree a bit and didn't see any bugs fall out. Looks great :). Glad you were able to get to this.

**Pull request merged**